Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -4641,19 +4641,35 @@ Color LocalFrameView::documentBackgroundColor() const
     // Background images are unfortunately impractical to include.
 
     // Return invalid Color objects whenever there is insufficient information.
-    if (!m_frame->document())
+    auto* document = m_frame->document();
+    if (!document)
         return Color();
 
-    auto* htmlElement = m_frame->document()->documentElement();
-    auto* bodyElement = m_frame->document()->bodyOrFrameset();
+    if (auto* fullscreenDocument = document->fullscreenManager().deepestFullscreenDocument())
+        document = fullscreenDocument;
```

Comment:
It makes me very nervous to reassign `document` to a document from another, potentially cross-origin frame. Also, with site isolation, there is no way this can work. I think this code has to be different.

Reply:
The caller of LocalFrameView::documentBackgroundColor() is going to have to call something like `outermostFullscreenView()->documentBackgroundColor()` rather than `mainFrameView()->documentBackgroundColor()`. In which case, this existing code could work without modification.

But for the `document` reassignment, I renamed it `backgroundDocument` and moved its initializer into a lambda to hopefully clarify that no documents are being re-assigned here.